app/controllers/samples_reports_controller.rb

```diff
@@ -0,0 +1,158 @@
+    @samples_reports = @samples_reports.joins("LEFT JOIN samples_report_samples ON samples_report_samples.samples_report_id = samples_reports.id 
+                                               LEFT JOIN samples ON samples_report_samples.sample_id = samples.id 
+                                               LEFT JOIN boxes ON boxes.id = samples.box_id")
+                                        .where("boxes.uuid LIKE concat('%', ?, '%')", params[:box_uuid])
+                                        .group(:samples_report_id) unless params[:box_uuid].blank?
+    @samples_reports = @samples_reports.joins("LEFT JOIN samples_report_samples ON samples_report_samples.samples_report_id = samples_reports.id 
+                                        LEFT JOIN samples ON samples_report_samples.sample_id = samples.id 
+                                        LEFT JOIN batches ON batches.id = samples.batch_id")
+                                 .where("batches.batch_number LIKE concat('%', ?, '%')", params[:batch_number])
+                                 .group(:samples_report_id) unless params[:batch_number].blank?
+
+    #paginate samples report
+    @samples_reports = perform_pagination(@samples_reports)
+  end
+  
+  def new
+    @samples_report = SamplesReport.new({
+      institution: @navigation_context.institution,
+      site: @navigation_context.site
+    })
+
+    @boxes = []
+    
+    prepare_for_institution_and_authorize(@samples_report, CREATE_INSTITUTION_SAMPLES_REPORT)
+  end
+
+  def create
+    @institution = @navigation_context.institution
+    return unless authorize_resource(@institution, CREATE_INSTITUTION_SAMPLES_REPORT)
+
+    @samples_report = SamplesReport.new()
+    @samples_report.institution = @institution
+    @samples_report.site = @navigation_context.site
+    @samples_report.name = params[:samples_report][:name]
+    
+    samples_report_samples = []
+    @boxes = [] # For form reloading in case of validation error
+    if params[:samples_report][:boxes_attributes]
+      @boxes_p = params[:samples_report][:boxes_attributes]
+      @boxes_p.each do |box_param|
+        box_id = @boxes_p[box_param][:box_id]
+        box = Box.find_by_id(box_id)
+        box.samples.each do |sample|
+          samples_report_samples  << SamplesReportSample.new(samples_report: @samples_report, sample: sample)
+        end
+      end
+    end
+
+    @samples_report.samples_report_samples = samples_report_samples
+    
+    if @samples_report.save
+      redirect_to samples_reports_path, notice: 'Box report was successfully created.'
+    else
+      render action: 'new'
+    end
+  end
+
+  def show
+    @samples_report = SamplesReport.find_by_id(params[:id])
+    @reports_data = measured_signal_data(@samples_report)
+    @samples_without_results_count = @samples_report.samples.where("core_fields NOT LIKE '%measured_signal%'").count
```

Comment:
Is this OK?

Reply:
We might be able to. We should be able to use the following:

```ruby
.where('core_fields->>"$.measured_signals" IS NOT NULL')
```

To be efficient we'd have to create an ugly [virtual generated column to create an index](https://dev.mysql.com/doc/refman/5.7/en/create-table-secondary-indexes.html#json-column-indirect-index), but given how many samples there are a sequential scan might not be an issue.

Reply:
I said :poop: the core fields are... encoded in YAML :sob: